### PR TITLE
PLANET-7542: Move e2e tests from the plugin to the theme

### PR DESCRIPTION
**Pull request by GP-Dan-Tovbein**
Ref: https://jira.greenpeace.org/browse/PLANET-7542

# Description
Changed the initial scope for the ticket. The new requirement is to move all missing specs from plugin. Indeed migrations if they are needed.

I created a new directory `test/blocks` and include all block specs instead of the root directory which is how it's now added into the plugin. Also created the `Admin` rest directory.

You can see the list of blocks that were already moved into the theme as a part of the `monorepo` epic.

## Testing
From dev, you can run `npm run env:e2e`
Or indivudally through `npx playwright test tests/e2e/blocks/{{filename}}.spec.js`

### Updates ⚠️ 
- Campaign covers specs are removed from this PR since will be covered by [PLANET-7489](https://jira.greenpeace.org/browse/PLANET-7489)

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [umbriel](https://www-dev.greenpeace.org/test-umbriel) | [admin](https://www-dev.greenpeace.org/test-umbriel/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-umbriel/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-umbriel) | [composer-local.json](https://github.com/greenpeace/planet4-test-umbriel/blob/main/composer-local.json)

:watch: 2024.07.19 14:11:05

**Review by mardelnet — COMMENTED** [review]
Nice job, @dantovbein 
Before begin checking this, I have a couple of questions:
- Is there any related PR in the plugin repository with the removed files?
- Can you explain further how to run the tests from `dev` using `npm run env:e2e`?
- I see some tests are failing right now, would you mind checking them?

**Comment by GP-Dan-Tovbein** [comment]
Hey @mardelnet ! Here are my responses:

> * Is there any related PR in the plugin repository with the removed files?

These files are still in the plugin

> * Can you explain further how to run the tests from `dev` using `npm run env:e2e`?

I prefer to use isolated commands since the one we have within the scripts is also installing
`npx playwright test tests/e2e/[filename].js`

> * I see some tests are failing right now, would you mind checking them?

Just updated!

**Comment by mardelnet** [comment]
Thanks @dantovbein 

> I prefer to use isolated commands since the one we have within the scripts is also installing `npx playwright test tests/e2e/[filename].js`

In the PR description, you mention this command has to be run on `dev`, not in the local environment. My question is: how can I run it on `dev`?

> Just updated!

I still see some tests failing.

**Comment by GP-Dan-Tovbein** [comment]
> Thanks @dantovbein
> 
> > I prefer to use isolated commands since the one we have within the scripts is also installing `npx playwright test tests/e2e/[filename].js`
> 
> In the PR description, you mention this command has to be run on `dev`, not in the local environment. My question is: how can I run it on `dev`?
> 
> > Just updated!
> 
> I still see some tests failing.

@mardelnet all specs are passing now! Could you check it again, please? 🙏

**Review by mardelnet — COMMENTED** [review]
@dantovbein 
Should we also move the following items from the plugin?

- tests/e2e/articles.spec.js
- tests/e2e/blocks-report-api.spec.js
- tests/e2e/editor-selector.spec.js
- tests/e2e/page-layouts-modal.spec.js
- tests/e2e/related-articles.spec.js
- tests/e2e/split-two-columns.spec.js
- tests/e2e/spreadsheet.spec.js
- tests/e2e/take-action-boxout.spec.js

File: src/Admin/Transient.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+use WP_REST_Request;
+use WP_REST_Server;
+use WP_Error;
+
+/**
+ * Transient cache.
+ */
+class Transient
+{
+    private wpdb $db;
```
[the remark this answers is not in the document]
Comment (GP-Dan-Tovbein) [reply src/Admin/Transient.php:18]: Yes, [here](https://github.com/greenpeace/planet4-master-theme/pull/2326/files/b927f87d7faf19fa14dede5275628d0666cbeca3#diff-c45aed64931d7a695d56b965b5d91318a52863ac7131b78759eed11e6ec6564bR30). It is a private class variable.

File: tests/e2e/tools/lib/covers.js
[the remark this answers is not in the document]
Comment (GP-Dan-Tovbein) [reply tests/e2e/tools/lib/covers.js:1]: Just removed! Let's wait for https://jira.greenpeace.org/browse/PLANET-7489

File: src/Admin/Rest.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * Initialize class if all checks are ok.
+     */
+    public function load(): void
+    {
+        add_action('rest_api_init', function (): void {
+            /**
+             * Access to transient cache for admin purposes.
+             */
+            register_rest_route(
```
[the remark this answers is not in the document]
Comment (GP-Dan-Tovbein) [reply src/Admin/Rest.php:25]: This wouldn't affect its functionality

File: src/Admin/Transient.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mardelnet) [reply src/Admin/Transient.php:18]: You're right, however, we are not using it (not `global $wpdb` either).
But I suppose it doesn't harm to keep it.

**Review by mardelnet — APPROVED** [review]
Great job @dantovbein

**Pull request merged**